File: medicine/src/main/java/com/sih/test/Injection.java
package com.sih.test;

import android.content.Context;
import androidx.annotation.NonNull;

import com.sih.test.data.source.MedicineRepository;
import com.sih.test.data.source.local.MedicinesLocalDataSource;

/**
 * Created by gautam on 13/07/17.
 */

public class Injection {

    public static MedicineRepository provideMedicineRepository(@NonNull Context context) {
        return MedicineRepository.getInstance(MedicinesLocalDataSource.getInstance(context));
    }
}

File: app/src/main/java/com/sih/eldify/medicines/Injection.kt
package com.sih.eldify.medicines

import android.content.Context

import com.sih.eldify.medicines.data.source.MedicineRepository
import com.sih.eldify.medicines.data.source.local.MedicinesLocalDataSource

/**
 * Created by gautam on 13/07/17.
 */
object Injection {
    fun provideMedicineRepository(context: Context): MedicineRepository {
        return MedicineRepository.getInstance(MedicinesLocalDataSource.getInstance(context)!!)!!
    }
}